var/spack/repos/builtin/packages/qmcpack/package.py

```diff
@@ -323,79 +336,55 @@ def cmake_args(self):
         # Next two environment variables were introduced in QMCPACK 3.5.0
         # Prior to v3.5.0, these lines should be benign but CMake
         # may issue a warning.
-        if 'intel-mkl' in spec:
+        if '^intel-mkl' in spec:
```

Comment:
```suggestion
        if '^mkl' in spec:
```
Both `intel-mkl` and `intel-parallel-studio+mkl` provide the `mkl` virtual dependency.

Reply:
A similar change was implemented in QE, but seems to be causing a problem. I just filed an issue about it:
https://github.com/spack/spack/issues/15613